### PR TITLE
zeek: 2.5.5 -> 3.0.0

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change

superseeds and closes https://github.com/NixOS/nixpkgs/pull/62695

---
feedback from @tobim in #62695:
>Add caf to buildInputs and "-DCAF_ROOT_DIR=${caf}" to cmakeFlags, otherwise the internal version will be used adding unnecessary compilation time.  

unfortunately, I can't get it to built with the `caf` from nixpkgs  
> Add rocksdb to buildInputs to enable brokers store module.

done
> The broker python bindings require the ipaddress module which was introduced in python-3.3, so the python parameter should be switched to python3.

Not necessary, python2 users can still install python-ipaddress from pypi
>It would be nice to have a toPythonModule entry in pythonPackages.nix, otherwise the module will be hidden from users.

Since I'm not familiar with the toPythonModule behavior, I’m going to tackle this later
>The geoip dependency can be replaced with libmaxminddb, this also enables a requirement for ncurses in one of the tests.

done

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @tobim, I'd appreciate it you could review this PR

### CODE REVIEW

File: pkgs/applications/networking/ids/zeek/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{stdenv, fetchurl, cmake, flex, bison, openssl, libpcap, perl, zlib, file, curl
```
Comment (tobim) [inline pkgs/applications/networking/ids/zeek/default.nix]: `perl` should not be necessary.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -1196,9 +1196,7 @@ in
 
   brltty = callPackage ../tools/misc/brltty { };
 
-  bro = callPackage ../applications/networking/ids/bro {
-    openssl = openssl_1_0_2;
-  };
+  zeek = callPackage ../applications/networking/ids/zeek { };
```
Comment (tobim) [inline pkgs/top-level/all-packages.nix]: nit: It would be nice to retain the alphabetical order.

**Review by tobim — CHANGES_REQUESTED** [review]
Looks good! I did some light testing and the main executable worked as expected.


> >    Add caf to buildInputs and "-DCAF_ROOT_DIR=${caf}" to cmakeFlags, otherwise the internal version will be used adding unnecessary compilation time.
>
> unfortunately, I can't get it to built with the caf from nixpkgs

I think that is because `caf` got updated to 0.17 in the meantime, but `zeek` is still on 0.16..4

You can add to the list of maintainers.

**Comment by tobim** [comment]
Sorry I meant to say add me as maintainer.

**Comment by marsam** [comment]
@tobim thanks for reviewing. I've addressed your comments

**Comment by marsam** [comment]
no problem, I added you as maintainer

**Review by tobim — APPROVED** [review]

**Comment by tobim** [comment]
I'll try to get around to write a function to build with packages from https://github.com/zeek/packages soon.

**Review by tobim — APPROVED** [review]

**Pull request merged**